cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -55,63 +54,48 @@
   private final NamespaceQueryAdmin namespaceQueryAdmin;
   private final CConfiguration cConf;
 
-  public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
-                            PropertiesResolver propertiesResolver, ListeningExecutorService taskExecutor,
+  public ScheduleTaskRunner(ProgramLifecycleService lifecycleService, PropertiesResolver propertiesResolver,
+                            ListeningExecutorService taskExecutor,
                             NamespaceQueryAdmin namespaceQueryAdmin, CConfiguration cConf) {
-    this.store = store;
     this.lifecycleService = lifecycleService;
     this.propertiesResolver = propertiesResolver;
     this.executorService = taskExecutor;
-    this.requirementsChecker = new RunConstraintsChecker(store);
     this.namespaceQueryAdmin = namespaceQueryAdmin;
     this.cConf = cConf;
   }
 
-  /**
-   * Checks if all schedule requirements are satisfied,
-   * then executes the given program without blocking until its completion.
-   *
-   * @param programId Program Id
-   * @param systemOverrides Arguments that would be supplied as system runtime arguments for the program.
-   * @param userOverrides Arguments to add to the user runtime arguments for the program.
-   * @return a {@link ListenableFuture} object that completes when the program completes
-   * @throws TaskExecutionException if program is already running or program is not found.
-   * @throws IOException if program failed to start.
-   */
-  public ListenableFuture<?> run(ProgramId programId, Map<String, String> systemOverrides,
-                                 Map<String, String> userOverrides) throws Exception {
+  public void launch(Job job) throws Exception {
+    ProgramSchedule schedule = job.getSchedule();
+    ProgramId programId = schedule.getProgramId();
     Map<String, String> userArgs = Maps.newHashMap();
     Map<String, String> systemArgs = Maps.newHashMap();
-
-    String scheduleName = systemOverrides.get(ProgramOptionConstants.SCHEDULE_NAME);
-    ApplicationSpecification appSpec = store.getApplication(programId.getParent());
-    if (appSpec == null || appSpec.getSchedules().get(scheduleName) == null) {
-      throw new TaskExecutionException(String.format(UserMessages.getMessage(UserErrors.PROGRAM_NOT_FOUND), programId),
-                                       false);
-    }
-
-    ScheduleSpecification spec = appSpec.getSchedules().get(scheduleName);
-    if (!requirementsChecker.checkSatisfied(programId, spec.getSchedule())) {
-      return Futures.<Void>immediateFuture(null);
+    Map<String, String> notificationProperties = job.getNotifications().get(0).getProperties();
+    String userOverridesString = notificationProperties.get(ProgramOptionConstants.USER_OVERRIDES);
+    if (userOverridesString != null) {
+      Map<String, String> userOverrides = GSON.fromJson(userOverridesString, STRING_STRING_MAP);
+      userArgs.putAll(userOverrides);
```

Comment:
we still want schedule properties to override both userOverrides as well as `propertiesResolver.getUserProperties(programId.toId())`, right?